File: content-by-language/ruby/outbound-integration.rb
# We recommend using the 'curb' library because the default
# net/http library does not support connecting to a proxy using TLS.
require 'curb'
require 'json'

proxy = 'https://{ACCESS_CREDENTIALS}@{VAULT_HOST}:{SECURE_PORT}'
uri = '{VGS_SAMPLE_ECHO_SERVER}/post'

c = Curl::Easy.new(uri) do |http|
  http.headers["Content-Type"] = "application/json"
  http.cacert = '{CERT_LOCATION}'
  http.proxy_url = proxy
  http.ssl_verify_peer = true
  http.post_body = {account_number: '{ALIAS}'}.to_json
  http.post
end

puts "Response #{c.status}: #{c.body}"

File: content-by-language/go/outbound-integration.go
package main

import (
  "bytes"
  "encoding/json"
  "io/ioutil"
  "fmt"
  "net/http"
  "os"
  "crypto/tls"
  "crypto/x509"
)


type Payload struct {
  Account string `json:"account_number"`
}

func main() {
  // You can set the proxy as an HTTPS env variable proxyUrl and go will use by default:
  os.Setenv("HTTPS_PROXY", "{SECURE_PROTOCOL}://{ACCESS_CREDENTIALS}@{VAULT_HOST}:{SECURE_PORT}")

  data := Payload{
    Account: "{ALIAS}",
  }
  payloadBytes, err := json.Marshal(data)
  if err != nil {
    fmt.Println(err)
  }

  body := bytes.NewReader(payloadBytes)

  caCert, err := ioutil.ReadFile("{CERT_LOCATION}")
  if err != nil {
    fmt.Println(err)
  }
  caCertPool := x509.NewCertPool()
  caCertPool.AppendCertsFromPEM(caCert)

  client := &http.Client{
    Transport: &http.Transport{
      Proxy: http.ProxyFromEnvironment,
      TLSClientConfig: &tls.Config{
        RootCAs: caCertPool,
        InsecureSkipVerify: false,
      },
    },
  }

  req, err := http.NewRequest("POST", "{VGS_SAMPLE_ECHO_SERVER}/post", body)
  if err != nil {
    fmt.Println(err)
  }
  req.Header.Set("Content-Type", "application/json")

  resp, err := client.Do(req)
  if err != nil {
    fmt.Println(err)
  }

  defer resp.Body.Close()

  respB, err := ioutil.ReadAll(resp.Body)
  if err != nil {
    fmt.Println(err)
  }
  fmt.Println(string(respB))
}
